ximage/tests/labels/test_labels.py

```diff
@@ -0,0 +1,736 @@
+
+def test_get_labels_with_requested_occurrence():
+    """Test _get_labels_with_requested_occurrence"""
+
+    array = np.array(
+        [
+            [0, 1, 2],
+            [2, 3, 3],
+            [3, 0, 0],
+        ]
+    )
+
+    label_indices = labels._get_labels_with_requested_occurrence(array, 1, 3)
+    assert np.array_equal(label_indices, np.array([1, 2, 3]))
+
+    # Test no labels found
+    label_indices = labels._get_labels_with_requested_occurrence(array, 4, 4)
+    assert np.array_equal(label_indices, np.array([]))
+
+
+def test_ensure_valid_label_arr():
+    """Test _ensure_valid_label_arr"""
+
+    array = np.array([[float("nan"), 0, 1]])
+    validated_array = labels._ensure_valid_label_arr(array)
+    assert np.array_equal(validated_array, np.array([[0, 0, 1]]))
+
+    # Test invalid values (negative, non-integer)
+    invalid_values = [-1, 0.5]
+    for invalid_value in invalid_values:
+        array = np.array([[invalid_value, 0, 1]])
+        with pytest.raises(ValueError):
+            labels._ensure_valid_label_arr(array)
+
+
+def test_ensure_valid_label_indices():
+    """Test _ensure_valid_label_indices"""
+
+    label_indices = np.array([float("nan"), 0, 1, 2, 3])
+    label_indices_valid = labels._ensure_valid_label_indices(label_indices)
+    assert np.array_equal(label_indices_valid, np.array([1, 2, 3]))
+
+
+def test_check_unique_label_indices():
+    """Test _check_unique_label_indices"""
+
+    labels._check_unique_label_indices(np.array([1, 2, 3]))
+
+    with pytest.raises(ValueError):
+        labels._check_unique_label_indices(np.array([1, 2, 2]))
+
+
+def test_get_new_label_value_dict():
+    """Test _get_new_label_value_dict"""
+
+    label_indices = np.array([1, 2, 3])
+    max_label = 4
+    expected_value_dict = {
+        0: 0,
+        1: 1,
+        2: 2,
+        3: 3,
+        4: 0,
+    }
+
+    new_label_value_dict = labels._get_new_label_value_dict(label_indices, max_label)
+    assert new_label_value_dict == expected_value_dict
+
+
+def test_np_redefine_label_array():
+    """Test _np_redefine_label_array"""
+
+    array = np.array(
+        [
+            [0, 1, 2],
+            [2, 3, 3],
+            [3, 0, 0],
+        ]
+    )
+
+    # Call without providing label_indices
+    array_returned = labels._np_redefine_label_array(array)
+    assert np.array_equal(array_returned, array)
+
+    # Call with filtering label_indices
+    label_indices = np.array([2, 3])
+    array_expected = np.array(
+        [
+            [0, 0, 1],
+            [1, 2, 2],
+            [2, 0, 0],
+        ]
+    )
+    array_returned = labels._np_redefine_label_array(array, label_indices)
+    assert np.array_equal(array_returned, array_expected)
+
+    # Check exception raised if label_indices is empty
+    with pytest.raises(ValueError):
+        labels._np_redefine_label_array(array, np.array([]))
+
+
+def test_xr_redefine_label_array():
+    """Test _xr_redefine_label_array"""
+
+    array = xr.DataArray(
```

Comment:
This array is being commonly used, I would suggest defining this as a fixture in `conftest.py`, and even better labelling it something nice if it represents a common array to be used.

```
@pytest.fixture
def common_3_by_3_matrix() -> xr.DataArray:
        return  xr.DataArray(
        [
            [0, 1, 2],
            [2, 3, 3],
            [3, 0, 0],
        ]
    )
```

then:
```
def test_xr_redefine_label_array(common_3_by_3_matrix: xr.DataArray):
    ... 
```

Reply:
I thought about it, but then I found myself creating several different matrices for testing different features. Also, I find it nice to directly see what the matrix looks like in the test. What do you think?

Reply:
For me fine to just define close to the test. 
Maybe you can define a fixture for an array representing edge cases: i.e all zeros, all.nan, all 1 ... 
I don't have a strong feeling about that.

BTW: If you use fixtures, would be good if at the top of the file we list the fixture that we use ;)

Reply:
Both approaches are for me. If the array is defining a feature of something specific that is common to all tests, then it would be useful for it to be a fixture.